src/traces/bar/hover.js

```diff
@@ -34,6 +34,7 @@ function hoverOnBars(pointData, xval, yval, hovermode) {
     var trace = cd[0].trace;
     var t = cd[0].t;
     var isClosest = (hovermode === 'closest');
+    var isWaterfall = (trace.type === 'waterfall');
```

Comment:
Hmm. I guess it would be annoying to move this logic in `waterfall/hover.js`. :disappointed: